ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
     
-    // Wait for LIS reboot.
-    nrf_delay_ms(10);
+    nrf_delay_ms(10); // Wait for LIS reboot.
     // Enable XYZ axes.
     lis2dh12_enable();
     lis2dh12_set_scale(LIS2DH12_SCALE);
-    // Sample rate 10 for activity detection.
     lis2dh12_set_sample_rate(LIS2DH12_SAMPLERATE_RAW);
     lis2dh12_set_resolution(LIS2DH12_RESOLUTION);
 
     lis2dh12_set_activity_interrupt_pin_2(LIS2DH12_ACTIVITY_THRESHOLD);
+    NRF_LOG_DEBUG("Accelerameter configuration done \r\n");                  // HOW TO init_status |= ACCELEROMETER_FAILED_INIT;  (_)??
 
-
-
-    // Setup BME280 - oversampling must be set for each used sensor.
-    bme280_set_oversampling_hum(BME280_HUMIDITY_OVERSAMPLING);
-    bme280_set_oversampling_temp(BME280_TEMPERATURE_OVERSAMPLING);
+            // oversampling must be set for each used sensor.
+    bme280_set_oversampling_hum  (BME280_HUMIDITY_OVERSAMPLING);
+    bme280_set_oversampling_temp (BME280_TEMPERATURE_OVERSAMPLING);
     bme280_set_oversampling_press(BME280_PRESSURE_OVERSAMPLING);
     bme280_set_iir(BME280_IIR);
     bme280_set_interval(BME280_DELAY);
     bme280_set_mode(BME280_MODE_NORMAL);
-    NRF_LOG_DEBUG("BME280 configuration done\r\n");
-  }
-
-  // Visually display init status. Hangs if there was an error, waits 3 seconds on success.
-  init_blink_status(err_code);
-
-  nrf_gpio_pin_set(LED_RED);  // Turn RED led off.
-  // Turn green led on to signal model +
-  // LED will be turned off in power_manage.
-  if (model_plus) { nrf_gpio_pin_clear(LED_GREEN); }
-
-  // Delay before advertising so we get valid data on first packet
-  nrf_delay_ms(MAIN_LOOP_INTERVAL_RAW + 100);
-
-  // Init ok, start watchdog with default wdt event handler (reset).
-  init_watchdog(NULL);
-
-  // start advertising
-  bluetooth_advertising_start();
-
-  // Enter main loop.
+    NRF_LOG_DEBUG("BME280 configuration done \r\n");                            // HOW TO init_status |= BME_FAILED_INIT; (_)
+  }   
+
+    // show to production QA and anyone else who's watching
+    if (init_status ) { NRF_LOG_WARNING (" -- Initalization error :  %d \r\n", init_status);
+                        for ( int16_t i=0; i<13; i++){ RED_LED_ON; nrf_delay_ms(100u); RED_LED_OFF; nrf_delay_ms(200u); }
+                        nrf_delay_ms(1000u);  }
+ 
+
+  if (model_plus) GREEN_LED_ON;    // will be turned off in power_manage.
+
+  nrf_delay_ms(MAIN_LOOP_INTERVAL_RAW + 100); // Delay before advertising so we get valid data on first packet
+
+  bluetooth_advertising_start(); NRF_LOG_INFO("Advertising start  \r\n");           // can this fail ?
+
+  if (init_status) {     // Report on status of initalization 
+   url_buffer[9]=init_status>>8;url_buffer[10]=init_status & 0xFF;       //  URL ruu.vi/#xx
+   bluetooth_set_eddystone_url(url_buffer, sizeof(url_buffer) ); nrf_delay_ms(500u); // keep sending this for a while
+//  raspberry pi:  

```

Comment:
Raspberry Pi specific comments could rather be a link to instructions in case there is an update.

Reply:
I will revise the Raspberry Pi comment. Where should the link point?

Reply:
Is there a source with the same information somewhere?

Reply:
In case there is not, I'd recommend a "tips and tricks" md file in the ruuvi-sensor-protocols repository, as the hcidump paired with grep is very useful in inspecting/testing the different data formats being transmitted, which is not strictly limited to ruuvi_firmware only